piccolo_api/crud/endpoints.py

```diff
@@ -806,18 +807,32 @@ async def post_single(
         except ValidationError as exception:
             return Response(str(exception), status_code=400)
 
-        try:
-            row = self.table(**model.dict())
-            if self._hook_map:
-                row = await execute_post_hooks(
-                    hooks=self._hook_map, hook_type=HookType.pre_save, row=row
+        if issubclass(self.table, BaseUser):
+            self.table._validate_password(password=cleaned_data["password"])
+            cleaned_data["password"] = self.table.hash_password(
+                cleaned_data["password"]
+            )
+            model = self.pydantic_model(**cleaned_data)
+            user_row = self.table(**model.dict())
+            await user_row.save().run()
+            return Response(status_code=200)
```

Comment:
I see your idea. You might be able to do this instead:

```python
user = await self.table.create_user(**model.dict())
json = dump_json({'id': user.id})
return CustomJSONResponse(json, status_code=201)
```

It still validates the password and hashes it. Also, it returns the data in the same format as other tables.

Reply:
This works perfectly and is much simpler. Thanks.